tests/utils/test_dataset_creation.py

```diff
@@ -171,6 +312,10 @@ def test_generate_dataset_with_external_buffer(dataset_id, env_id):
     assert len(dataset.episode_indices) == num_episodes
 
     check_data_integrity(dataset._data, dataset.episode_indices)
+    check_episode_data_integrity(
+        dataset, dataset.spec.observation_space, dataset.spec.action_space
+    )
+
```

Comment:
What's the difference between `check_data_integrity`and `check_episode_data_integrity`? It seems they do more or less the same thing, one with MinariStorage and one with MinariDataset? Can we merge them and check just once?

Reply:
I think this might be a good idea, but it does depend on if we want backend specific tests, and I think it should be in a future PR.